File: src/main/java/com/ding/aiplatjava/service/PromptService.java
package com.ding.aiplatjava.service;

import java.util.List;

import com.ding.aiplatjava.entity.Prompt;

/**
 * Prompt 服务接口
 * 定义 Prompt 相关的业务逻辑操作
 */
public interface PromptService {

    /**
     * 根据ID获取Prompt
     * @param id Prompt ID
     * @param userId 当前用户ID (用于权限校验)
     * @return Prompt实体，如果不存在或无权访问返回null
     */
    Prompt getPromptById(Long id, Long userId);

    /**
     * 获取指定用户的所有Prompt
     * @param userId 用户ID
     * @return Prompt列表
     */
    List<Prompt> getPromptsByUserId(Long userId);

    /**
     * 创建新的Prompt
     * @param prompt 要创建的Prompt实体 (应包含userId, title, content, category)
     * @param userId 当前用户ID (用于设置userId字段)
     * @return 创建成功后的Prompt实体 (包含生成的ID和时间戳)
     */
    Prompt createPrompt(Prompt prompt, Long userId);

    /**
     * 更新现有的Prompt
     * @param id 要更新的Prompt ID
     * @param prompt 包含更新信息的Prompt实体 (允许更新 title, content, category)
     * @param userId 当前用户ID (用于权限校验)
     * @return 更新成功后的Prompt实体，如果不存在或无权访问返回null
     */
    Prompt updatePrompt(Long id, Prompt prompt, Long userId);

    /**
     * 删除Prompt
     * @param id 要删除的Prompt ID
     * @param userId 当前用户ID (用于权限校验)
     * @return 删除成功返回true，否则返回false
     */
    boolean deletePrompt(Long id, Long userId);
} 

File: my-ai-platform-frontend/src/services/promptService.ts
import apiClient from './apiClient';

// --- 类型定义 ---

/**
 * Prompt 数据接口 (对应后端的 PromptDto 或 Prompt 实体)
 * 注意：时间戳可能是字符串 (来自 DTO) 或 Date 对象 (如果直接解析实体)
 *       但为简单起见，我们统一处理为字符串或 null。
 */
export interface Prompt {
  id: number | string;
  title: string;
  content: string;
  category: string | null; // 分类可以是 null
  createdAt?: string | null; // DTO 是格式化字符串
  updatedAt?: string | null; // DTO 是格式化字符串
  // userId?: number | string; // 实体包含，DTO 不含，通常前端不需要
}

/**
 * 用于创建或更新 Prompt 时发送的数据接口。
 * 只包含用户需要编辑的字段。
 */
export interface PromptData {
  title: string;
  content: string;
  category: string | null;
}

// --- 服务方法 ---

/**
 * 获取当前用户的所有 Prompts。
 * @returns {Promise<Prompt[]>} 一个 Promise，解析为 Prompt 数组。
 */
export const getPrompts = async (): Promise<Prompt[]> => {
  console.log('[promptService] Fetching all prompts...');
  try {
    // 后端 GET /api/prompts 返回 List<PromptDto>
    const response = await apiClient.get<Prompt[]>('/api/prompts');
    console.log('[promptService] Fetched prompts:', response.data);
    return response.data;
  } catch (error) {
    console.error('[promptService] Failed to fetch prompts:', error);
    throw new Error('获取 Prompt 列表失败');
  }
};

/**
 * 根据 ID 获取单个 Prompt。
 * @param {number | string} id - Prompt ID。
 * @returns {Promise<Prompt>} 一个 Promise，解析为单个 Prompt 对象。
 */
export const getPromptById = async (id: number | string): Promise<Prompt> => {
  console.log(`[promptService] Fetching prompt by ID: ${id}`);
  try {
    // 后端 GET /api/prompts/{id} 返回 Prompt 实体
    const response = await apiClient.get<Prompt>(`/api/prompts/${id}`);
    console.log('[promptService] Fetched prompt:', response.data);
    return response.data;
  } catch (error) {
    console.error(`[promptService] Failed to fetch prompt ${id}:`, error);
    throw new Error(`获取 Prompt (ID: ${id}) 失败`);
  }
};

/**
 * 创建新的 Prompt。
 * @param {PromptData} data - 要创建的 Prompt 数据 (title, content, category)。
 * @returns {Promise<Prompt>} 一个 Promise，解析为创建后的 Prompt 对象 (后端返回实体)。
 */
export const createPrompt = async (data: PromptData): Promise<Prompt> => {
  console.log('[promptService] Creating new prompt:', data);
  try {
    // 后端 POST /api/prompts 请求体接收 Prompt，返回创建后的 Prompt 实体
    const response = await apiClient.post<Prompt>('/api/prompts', data);
    console.log('[promptService] Created prompt:', response.data);
    return response.data;
  } catch (error) {
    console.error('[promptService] Failed to create prompt:', error);
    throw new Error('创建 Prompt 失败');
  }
};

/**
 * 更新现有的 Prompt。
 * @param {number | string} id - 要更新的 Prompt ID。
 * @param {PromptData} data - 更新后的 Prompt 数据 (title, content, category)。
 * @returns {Promise<Prompt>} 一个 Promise，解析为更新后的 Prompt 对象 (后端返回实体)。
 */
export const updatePrompt = async (id: number | string, data: PromptData): Promise<Prompt> => {
  console.log(`[promptService] Updating prompt ${id}:`, data);
  try {
    // 后端 PUT /api/prompts/{id} 请求体接收 Prompt，返回更新后的 Prompt 实体
    const response = await apiClient.put<Prompt>(`/api/prompts/${id}`, data);
    console.log('[promptService] Updated prompt:', response.data);
    return response.data;
  } catch (error) {
    console.error(`[promptService] Failed to update prompt ${id}:`, error);
    throw new Error(`更新 Prompt (ID: ${id}) 失败`);
  }
};

/**
 * 删除 Prompt。
 * @param {number | string} id - 要删除的 Prompt ID。
 * @returns {Promise<void>} 一个 Promise，在删除成功时 resolve。
 */
export const deletePrompt = async (id: number | string): Promise<void> => {
  console.log(`[promptService] Deleting prompt ${id}`);
  try {
    // 后端 DELETE /api/prompts/{id} 成功返回 204 No Content
    await apiClient.delete(`/api/prompts/${id}`);
    console.log(`[promptService] Deleted prompt ${id}`);
  } catch (error) {
    console.error(`[promptService] Failed to delete prompt ${id}:`, error);
    throw new Error(`删除 Prompt (ID: ${id}) 失败`);
  }
}; 